File: src/problems_all/01_two_sum/twoSum.java
/*
Runtime:    2 ms
Memory:     45.03 MB
Submission: https://leetcode.com/problems/two-sum/submissions/1382250169/
*/

import java.util.HashMap;
import java.util.Arrays;

public class Main {
    public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numToIndex = new HashMap<>();
        int[] res = new int[2];

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (numToIndex.containsKey(complement)) {
                res[0] = numToIndex.get(complement);
                res[1] = i;
                return res;
            }
            numToIndex.put(nums[i], i);
        }

        return new int[0];
    }

    public static void main(String[] args) {
        int target = 9;
        int[] nums = {2, 7, 11, 15};

        int[] result = twoSum(nums, target);

        System.out.println(Arrays.toString(result));
    }
}

File: src/problems_all/01_two_sum/two_sum.cpp
/*
Runtime:    152 ms
Memory:     9.54 MB
Submission: https://leetcode.com/problems/two-sum/submissions/211518943/
*/

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int first, last;
        int length = nums.size();
        vector<int> v;
        for (int i = 0; i < length - 1; i++) {
            for (int j = i + 1; j < length; j++) {
                if (nums[i] + nums[j] == target) {
                    first = i;
                    last = j;
                    break;
                }
            }
        }
        v.push_back(first);
        v.push_back(last);

        return v;
    }
};

int main() {
    int target = 9;
    vector<int> nums {2, 7, 11, 15};
    vector<int> result;
    Solution solution;

    result = solution.twoSum(nums, target);
    cout << "[" << result[0] << ", " << result[1] << "]" << endl;

    return 0;
}